code/modules/mob/living/silicon/mommi/mommi_subtypes/sammi.dm

```diff
@@ -193,19 +183,13 @@
 
 /mob/living/silicon/robot/mommi/sammi/attack_ghost(var/mob/dead/observer/O)
 	if(!(src.key))
-		if(O.can_reenter_corpse)
-			var/response = alert(O,"Do you want to take it over?","This SAMMI has no soul","Yes","No")
-			if(response == "Yes")
-				if(!(src.key))
-					src.transfer_personality(O.client)
-					src.visible_message("<span class=\"warning\">[src] is connected to the SAMMI network!</span>")
-					icon_state = "sammi_online"
-					updateicon()
-				else if(src.key)
-					to_chat(src, "<span class='notice'>Someone has already began controlling this SAMMI. Try another! </span>")
-		else if(!(O.can_reenter_corpse))
```

Comment:
Why was the can_reenter_corpse check removed?

We have those to prevent somebody making something that can be taken over by a ghost, then topping themself to take it over by themselves.

Reply:
In testing I obliterated someone's corpse and they couldn't enter the sammi.

In play testing that flag never seemed to reset, and in a cursory look at the code I couldn't find where it ever reset through something like a cool down. 

Test coverage may be lacking here.

...also because sammis are so limited in their ability I felt like the abuse potential is low and thought they shouldn't care if the user could renter their corpse.

If that's the wrong stance to take here just let me know.